### PR TITLE
(WIP) (QENG-5715) Changes to support self-service Docker CI workflow

### PR DESCRIPTION
**Pull request by nwolfe**
**TL;DR** Make the final `WORKDIR` configurable during `build`

What's the best way to add support for this?

----

I ran into some incompatibilities when using this module as part of QE's upcoming "self-service Docker CI" workflow (QENG-5083).

This PR captures all the changes I ended up making in order to get the new workflow working.
I've already baked this module into an image that we plan to use in production: https://github.com/puppetlabs/docker-ci-tools/blob/master/dockerfiles/debian/jessie/jenkins/imagebuild/Dockerfile

Specifically, I had troubles with the `Dockerfile.erb` contents while trying to use our `universal` image as the `--from`, which is what we use to run all the Jenkins slaves with. See the `universal` image here: https://github.com/puppetlabs/docker-ci-tools/blob/master/dockerfiles/debian/jessie/jenkins/universal/Dockerfile

My issues were mostly surrounding the WORKDIR:
- `Dockerfile.erb` assumes/requires that `/` is the WORKDIR, yet
- I need the final WORKDIR of the image to be `/tmp`

In the interest of time, I simply hardcoded these WORKDIR directives into my fork of the module so I could continue.

The other problem I encountered was that Java 8 was being uninstalled by the `apt-get remove --purge -y wget` line. I assume this is probably a result of the `universal` image mostly, and how it's using or misusing apt to install java 8, because apt seems to think it's not needed and will thus delete it? Rather than figuring out and changing the `universal` image, it was easier for me to simply delete this `apt-get remove` line.

I ran into another issue regarding argument precedence between CLI and config file values, but I ended up not needing to use the `metadata.yaml` config file so this problem went away. You'll see a couple commits in this changeset though where I swapped the CLI vs file precedence, since it didn't use CLI values if they were also present in the config file.

CC @sbeaulie @smcelmurry @mchllweeks

### CODE REVIEW

**Comment by sbeaulie** [comment]
@garethr Do you have suggestions on how this repo should make the `WORKDIR` configurable?

**Comment by cdenneen** [comment]
WORKDIR is important for many node app containers so having this as a flag would be useful... for now just using Dockerfile so I can specify the WORKDIR instead.

👍

**Comment by cdenneen** [comment]
@nwolfe are we abandoning this?
